tools/statsnoop.py

```diff
@@ -114,20 +115,17 @@
 # system calls but the name of the actual entry point may
 # be different for which we must check if the entry points
 # actually exist before attaching the probes
-syscall_fnname = b.get_syscall_fnname("stat")
-if BPF.ksymname(syscall_fnname) != -1:
-    b.attach_kprobe(event=syscall_fnname, fn_name="syscall__entry")
-    b.attach_kretprobe(event=syscall_fnname, fn_name="trace_return")
-
-syscall_fnname = b.get_syscall_fnname("statfs")
-if BPF.ksymname(syscall_fnname) != -1:
-    b.attach_kprobe(event=syscall_fnname, fn_name="syscall__entry")
-    b.attach_kretprobe(event=syscall_fnname, fn_name="trace_return")
-
-syscall_fnname = b.get_syscall_fnname("newstat")
-if BPF.ksymname(syscall_fnname) != -1:
-    b.attach_kprobe(event=syscall_fnname, fn_name="syscall__entry")
-    b.attach_kretprobe(event=syscall_fnname, fn_name="trace_return")
+def try_attach_syscall_probes(syscall):
+    syscall_fnname = b.get_syscall_fnname(syscall)
+    if BPF.ksymname(syscall_fnname) != -1:
+        b.attach_kprobe(event=syscall_fnname, fn_name="syscall__entry")
+        b.attach_kretprobe(event=syscall_fnname, fn_name="trace_return")
+
+try_attach_syscall_probes("stat")
+try_attach_syscall_probes("statx")
```

Comment:
The first argument to statx is an `int`, not `const char *`.

Reply:
done, thanks.